server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -80,8 +80,11 @@ public FieldTypeLookup copyAndAddAll(String type,
             String aliasName = fieldAliasMapper.name();
             String path = fieldAliasMapper.path();
 
-            validateAlias(aliasName, path, aliases, fullName);
-            aliases = aliases.copyAndPut(aliasName, path);
+            String existingPath = aliases.get(aliasName);
+            if (Objects.equals(path, existingPath) == false) {
```

Comment:
nit: maybe using String.equals would be more common in this cases, but I'm fine with either way.

Reply:
I went with `Objects.equals` because it handles null values nicely (`existingPath` could be null).

Reply:
Makes sense, thanks for the explanation.